### PR TITLE
PERF: replace with list, closes #28084

### LINKED ISSUE #28084: pandas slow replace with int64 columns in dataframe
Reproduced issue in part without n2 and n3 and with n2 n3 see prior issue:
https://github.com/pandas-dev/pandas/issues/12257

The replace runs 10x slower when encountering int64 columns

WITHOUT n2,n3, **1.73s**
```
a1=np.zeros((40000000,1))
a1[:,:]=np.inf
df=pd.DataFrame(a1)
df['a1']=''
df['n1']=0.0
# df['n2']=0.0
# df['n2']=df['n2'].astype(np.int64)
# df['n3']=0.0
# df['n3']=df['n3'].astype(np.int64)
df['n1'].astype('datetime64[ns]')
df['a1']=df['a1'].astype('category')
%time df.replace([np.inf, -np.inf], np.nan)
```

WITH n2, n3 `21.4s`
```
a1=np.zeros((40000000,1))
a1[:,:]=np.inf
df=pd.DataFrame(a1)
df['a1']=''
df['n1']=0.0
# df['n2']=0.0
# df['n2']=df['n2'].astype(np.int64)
# df['n3']=0.0
# df['n3']=df['n3'].astype(np.int64)
df['n1'].astype('datetime64[ns]')
df['a1']=df['a1'].astype('category')
%time df.replace([np.inf, -np.inf], np.nan)
```

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [x] closes #28084
- [ ] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

based on the OP's example:

```
In [2]: df = pd.DataFrame({"A": 0, "B": 0}, index=range(4*10**7))

In [3]: %timeit df.replace([np.inf, -np.inf], np.nan)
5.18 s ± 423 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)    # <-- master
414 ms ± 7.19 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)   # <-- PR

In [4]: %timeit df.replace([np.inf, -np.inf], np.nan, inplace=True)
2.89 s ± 111 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)      # <-- master
69.6 µs ± 4 µs per loop (mean ± std. dev. of 7 runs, 10000 loops each)  # <-- PR

In [5]: %timeit df.replace([np.inf, -np.inf, 1], np.nan)
4.88 s ± 228 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)   # <-- master
466 ms ± 11.8 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)  # <-- PR
```

### CODE REVIEW

**Review by TomAugspurger — APPROVED** [review]
Probably worth adding an ASV.

**Comment by jbrockmendel** [comment]
good idea, will do

**Comment by jbrockmendel** [comment]
@WillAyd suggestions for how to make asv happy?

File: asv_bench/benchmarks/replace.py [hunk 1]
```diff
@@ -36,6 +36,23 @@ def time_replace_series(self, inplace):
         self.s.replace(self.to_rep, inplace=inplace)
 
 
+class ReplaceList:
+    # GH#28099
+
+    params = [(True, False)]
+    param_names = ["inplace"]
+
+    def setup_cache(self, inplace):
```
Comment (WillAyd) [inline asv_bench/benchmarks/replace.py]: I think having `inplace` here is a problem as `setup_cache` is not parametrized. Could remove that argument though I think better to just change to `setup` as this construction doesn't take that long

**Pull request merged**